qp_shotgun/humann2/humann2.py

```diff
@@ -84,50 +85,72 @@ def generate_humann2_analysis_commands(forward_seqs, reverse_seqs, map_file,
             % ', '.join(sn_by_rp.keys()))
 
     cmds = []
-    params = ' '.join(["%s %s" % (k, v) for k, v in viewitems(parameters)])
+    params = ['--%s "%s"' % (k, v) for k, v in viewitems(parameters) if v]
     for ffn, fn, s in samples:
         cmds.append('humann2 --input "%s" --output "%s" --output-basename '
                     '"%s" --output-format biom %s' % (ffn, join(out_dir, fn),
-                                                      s, params))
+                                                      s, ' '.join(params)))
 
     return cmds
 
 
-def get_sample_names_by_run_prefix(mapping_file):
-    """Generates a dictionary of run_prefix and sample names
+def humann2(qclient, job_id, parameters, out_dir):
+    """Run humann2 with the given parameters
```

Comment:
Change to HUMAnN2 per above... or is this referring to the command humann2 as opposed to the pipeline HUMAnN2?